tests/test_app/TestApp/Log/Engine/TestBaseLog.php

```diff
@@ -12,17 +12,27 @@
  */
 class TestBaseLog extends BaseLog
 {
+    /**
+     * @var string
+     */
+    protected $message = '';
+
     /**
      * Logs with an arbitrary level.
      *
      * @param mixed $level
      * @param string $message
      * @param array $context
      *
-     * @return string
+     * @return void
      */
     public function log($level, $message, array $context = [])
     {
-        return $this->_format($message, $context);
+        $this->message = $this->_format($message, $context);
+    }
+
+    public function getMessage()
```

Comment:
We should add string typehint here also for test methods. This avoids accidental usage as other type etc.